client/web-sveltekit/src/routes/[...repo=reporev]/-/compare/[...spec]/+page.svelte

```diff
@@ -0,0 +1,247 @@
+<script lang="ts">
+    // @sg EnableRollout
+    import { onMount } from 'svelte'
+
+    import { goto } from '$app/navigation'
+    import { resolveRoute } from '$app/paths'
+    import { page } from '$app/stores'
+    import Commit from '$lib/Commit.svelte'
+    import Icon from '$lib/Icon.svelte'
+    import LoadingSpinner from '$lib/LoadingSpinner.svelte'
+    import FileDiff from '$lib/repo/FileDiff.svelte'
+    import Scroller, { type Capture } from '$lib/Scroller.svelte'
+    import { TELEMETRY_RECORDER } from '$lib/telemetry'
+    import { createPromiseStore } from '$lib/utils'
+    import { Alert } from '$lib/wildcard'
+    import Button from '$lib/wildcard/Button.svelte'
+
+    import RepositoryRevPicker from '$lib/repo/RepositoryRevPicker.svelte'
+
+    import type { PageData, Snapshot } from './$types'
+
+    export let data: PageData
+
+    export const snapshot: Snapshot<{
+        scroller: Capture
+        diffs: ReturnType<NonNullable<typeof data.diffQuery>['capture']>
+    }> = {
+        capture() {
+            return {
+                scroller: scroller.capture(),
+                diffs: data.diffQuery?.capture(),
+            }
+        },
+        async restore({ scroller: scrollerData, diffs: diffsData }) {
+            await data.diffQuery?.restore(diffsData)
+            scroller.restore(scrollerData)
+        },
+    }
+
+    function handleSelect(baseRevision: string, headRevision: string): void {
+        goto(
+            resolveRoute('/[...repo=reporev]/-/compare/[...spec]', {
+                repo: $page.params.repo,
+                spec: `${baseRevision}...${headRevision}`,
+            })
+        )
+    }
+
+    function handleCommitPage(page: number): void {
+        const url = new URL($page.url)
+        url.searchParams.set('p', page.toString())
+        goto(url)
+    }
+
+    onMount(() => {
+        TELEMETRY_RECORDER.recordEvent('repo.compare', 'view')
+    })
+
+    let scroller: Scroller
+    let expandedDiffs = new Map<number, boolean>()
+    const commits = createPromiseStore<Awaited<typeof data.commits>>()
+
+    $: commits.set(data.commits)
```

Comment:
Any reason not to use the simpler construct? I guess the difference is that `commits` will continue to show the old value until the new value loads 🤔 Is that what we want?

```suggestion
    $: commits = toReadable(data.commits)
```

Reply:
> I guess the difference is that commits will continue to show the old value until the new value loads 🤔 Is that what we want?

I think so. It feels awkward when you click next/previous and the whole list disappears.